applications/nrf5340_audio/src/modules/audio_sync_timer.c

```diff
@@ -9,34 +9,112 @@
 
+#define AUDIO_SYNC_TIMER_NET_APP_IPC_EVT_CHANNEL 4
 #define AUDIO_SYNC_TIMER_NET_APP_IPC_EVT NRF_IPC_EVENT_RECEIVE_4
 
-const nrfx_timer_t audio_sync_timer_instance =
+static const nrfx_timer_t audio_sync_timer_instance =
 	NRFX_TIMER_INSTANCE(AUDIO_SYNC_TIMER_INSTANCE_NUMBER);
 
+#if !defined(CONFIG_AUDIO_SYNC_TIMER_USES_RTC)
 static uint8_t dppi_channel_timer_clear;
+#endif
 static uint8_t dppi_channel_i2s_frame_start;
 
+#if defined(CONFIG_AUDIO_SYNC_TIMER_USES_RTC)
+#define AUDIO_RTC_TIMER_INSTANCE_NUMBER                     0
+
+#define AUDIO_RTC_TIMER_I2S_FRAME_START_EVT_CAPTURE_CHANNEL 0
+#define AUDIO_RTC_TIMER_CURR_TIME_CAPTURE_CHANNEL           1
+#define AUDIO_RTC_TIMER_I2S_FRAME_START_EVT_CAPTURE         NRF_RTC_TASK_CAPTURE_0
+#define AUDIO_RTC_TIMER_CURR_TIME_CAPTURE_TASK              NRF_RTC_TASK_CAPTURE_1
+
+static const nrfx_rtc_config_t rtc_cfg = NRFX_RTC_DEFAULT_CONFIG;
+static const nrfx_rtc_t audio_rtc_timer_instance =
+	NRFX_RTC_INSTANCE(AUDIO_RTC_TIMER_INSTANCE_NUMBER);
+
+static uint8_t dppi_channel_timer_sync_with_rtc;
+static uint8_t dppi_channel_rtc_start;
+static volatile uint32_t num_rtc_overflows;
+#endif
+
 static nrfx_timer_config_t cfg = {.frequency = NRFX_MHZ_TO_HZ(1UL),
 				  .mode = NRF_TIMER_MODE_TIMER,
 				  .bit_width = NRF_TIMER_BIT_WIDTH_32,
 				  .interrupt_priority = NRFX_TIMER_DEFAULT_CONFIG_IRQ_PRIORITY,
 				  .p_context = NULL};
 
-static void event_handler(nrf_timer_event_t event_type, void *ctx)
+uint32_t audio_sync_timer_capture(void)
+{
+#if defined(CONFIG_AUDIO_SYNC_TIMER_USES_RTC)
+	nrf_rtc_task_trigger(audio_rtc_timer_instance.p_reg,
+			     AUDIO_RTC_TIMER_CURR_TIME_CAPTURE_TASK);
+	nrf_timer_task_trigger(NRF_TIMER1,
+		nrf_timer_capture_task_get(AUDIO_SYNC_TIMER_CURR_TIME_CAPTURE_CHANNEL));
+
+	uint32_t tick = nrf_rtc_cc_get(audio_rtc_timer_instance.p_reg,
+				       AUDIO_RTC_TIMER_CURR_TIME_CAPTURE_CHANNEL);
+	uint32_t remainder_us = nrf_timer_cc_get(NRF_TIMER1,
+						 AUDIO_SYNC_TIMER_CURR_TIME_CAPTURE_CHANNEL);
+
+	return ((tick * 30517578125UL) / 1000000000UL) +
+	       (num_rtc_overflows * 512000000UL) +
+	       remainder_us;
+#else
+	return nrfx_timer_capture(&audio_sync_timer_instance,
+				  AUDIO_SYNC_TIMER_CURR_TIME_CAPTURE_CHANNEL);
+#endif
+}
+
+uint32_t audio_sync_timer_capture_get(void)
 {
+#if defined(CONFIG_AUDIO_SYNC_TIMER_USES_RTC)
+	uint32_t remainder_us;
+	uint32_t tick;
+
+	tick = nrf_rtc_cc_get(audio_rtc_timer_instance.p_reg,
+			      AUDIO_RTC_TIMER_I2S_FRAME_START_EVT_CAPTURE_CHANNEL);
+	remainder_us = nrf_timer_cc_get(NRF_TIMER1,
+					AUDIO_SYNC_TIMER_I2S_FRAME_START_EVT_CAPTURE_CHANNEL);
+
+	return ((tick * 30517578125UL) / 1000000000UL) +
+	       (num_rtc_overflows * 512000000UL) +
+	       remainder_us;
+
+#else
+	return nrfx_timer_capture_get(&audio_sync_timer_instance,
+				      AUDIO_SYNC_TIMER_I2S_FRAME_START_EVT_CAPTURE_CHANNEL);
+#endif
 }
 
+static void unused_timer_isr_handler(nrf_timer_event_t event_type, void *ctx)
+{
+	(void)event_type;
```

Comment:
I suggest using the macro `ARG_UNUSED()` instead, as that is more used in our application at least